bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/discovery/BondDiscoveryService.java

```diff
@@ -0,0 +1,99 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.bondhome.internal.api.BondDevice;
+import org.openhab.binding.bondhome.internal.api.BondHttpApi;
+import org.openhab.binding.bondhome.internal.handler.BondBridgeHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class does discovery of discoverable things
+ *
+ * @author Arne Seime - Initial contribution
+ */
+@NonNullByDefault
+public class BondDiscoveryService extends AbstractDiscoveryService {
+    private static final long REFRESH_INTERVAL_MINUTES = 60;
+    public static final Set<ThingTypeUID> DISCOVERABLE_THING_TYPES_UIDS = SUPPORTED_DEVICE_TYPES;
+    private final Logger logger = LoggerFactory.getLogger(BondDiscoveryService.class);
+    private @Nullable ScheduledFuture<?> discoveryJob;
+    private final BondBridgeHandler bridgeHandler;
+    private BondHttpApi api;
+
+    public BondDiscoveryService(final BondBridgeHandler bridgeHandler) {
+        super(DISCOVERABLE_THING_TYPES_UIDS, 10);
+        this.bridgeHandler = bridgeHandler;
+        this.api = bridgeHandler.getBridgeAPI();
+        this.discoveryJob = null;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        discoveryJob = scheduler.scheduleWithFixedDelay(this::startScan, 0, REFRESH_INTERVAL_MINUTES, TimeUnit.MINUTES);
+    }
+
+    @Override
+    protected synchronized void startScan() {
+        logger.debug("Start scan for Bond devices.");
+        try {
+            final ThingUID bridgeUid = bridgeHandler.getThing().getUID();
+            api = bridgeHandler.getBridgeAPI();
+            List<String> deviceList = api.getDevices();
+            if (deviceList != null) {
+                for (final String deviceId : deviceList) {
+                    BondDevice thisDevice = api.getDevice(deviceId);
+                    if (thisDevice != null) {
+                        final ThingUID deviceUid = new ThingUID(thisDevice.type.getThingTypeUID(), bridgeUid, deviceId);
+                        final DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(deviceUid)
+                                .withBridge(bridgeUid).withLabel(thisDevice.name)
+                                .withProperty(CONFIG_DEVICE_ID, deviceId)
+                                .withProperty(PROPERTIES_DEVICE_NAME, thisDevice.name)
+                                .withRepresentationProperty(deviceId).build();
```

Comment:
The representation property must be the *name* of the property.
```suggestion
                                .withRepresentationProperty(CONFIG_DEVICE_ID).build();
```

Reply:
changed